test/metabase/driver/sql_jdbc/sync/describe_database_test.clj

```diff
@@ -208,3 +208,23 @@
                                   driver/*driver* conn schema (str table-name "_should_not_exist"))))
                      (is (true? (sql-jdbc.sync.interface/have-select-privilege?
                                  driver/*driver* conn schema table-name))))))))))))))
+
+(deftest sync-table-with-backslash-test
+  (mt/test-drivers #{:postgres} ;; TODO: fix and change this to test on (mt/sql-jdbc-drivers)
+    (testing "table with backslash in name, PKs, FKS are correctly synced"
+      (mt/with-temp-test-data [["human\\race"
+                                [{:field-name "humanraceid" :base-type :type/Integer :pk? true}
+                                 {:field-name "race" :base-type :type/Text}]
+                                [[1 "homo sapiens"]]]
+                               ["citizen"
+                                [{:field-name "citizen\\id" :base-type :type/Integer :pk? true}
+                                 {:field-name "race\\id" :base-type :type/Integer :fk "human\\race"}]
+                                [[1 1]]]]
+        (sync/sync-database! (mt/db))
```

Comment:
This is a redundant sync since `mt/with-temp-test-data` will run a metadata sync
```suggestion
```
see https://github.com/metabase/metabase/blob/6c7fbaf818ac13fc326447907ab5c723d77f2b5d/test/metabase/test/data/impl/get_or_create.clj#L122